src/util/get-body-size.ts

```diff
@@ -0,0 +1,42 @@
+import type { Readable } from 'node:stream';
+import { HttpBody } from '../types';
+import is from '@sindresorhus/is';
+
+export function getBodySize(
+  body: HttpBody | undefined,
+  headers: { [name: string]: string } | undefined,
+): number | undefined {
+  function isFormData(body: unknown): body is FormData {
+    return is.nodeStream(body) && is.function_((body as FormData).getBoundary);
+  }
+
+  if (headers && 'content-length' in headers) {
+    return Number(headers['content-length']);
+  }
+
+  if (!body) {
+    return 0;
+  }
+
+  if (is.string(body)) {
+    return Buffer.byteLength(body);
+  }
+
+  if (is.urlSearchParams(body)) {
+    return Buffer.byteLength(body.toString());
+  }
+
+  if (is.buffer(body)) {
+    return body.length;
+  }
+  if (isFormData(body)) {
+    return body.getLengthSync();
+  }
+
+  return undefined;
+}
```

Comment:
the only case not handled here is readable streams:
https://github.com/jsforce/jsforce/blob/5369d1d3e24bba0e384027d309c441c9ac5fa956/src/types/common.ts#L17

got stopped handling this in v12 too, see:
https://github.com/sindresorhus/got/issues/1486
https://github.com/sindresorhus/got/pull/1510